### DISCUSSION TITLE
Zero width spaces and xml entities

### DISCUSSION THREAD

**Discussion by luebbe** [WeblateOrg/weblate#6360] [Q&A] [answered]
We have one translation component where we intentionally allow (read want) zero width spaces as a sort of poor man's hyphenation. Weblate warns about the usage of zero width spaces. I guess I can turn off the check when I work my way through the docs.

* How can I insert a zero width space in weblate's editor other than copy&pasting it from somewhere else?
* Would it be possible for weblate to highlight zero width spaces in the editor, like it does for linebreaks?
* Would it be possible for weblate to preserve this character as xml entity?
   e.g. the string was uploaded as **_Ein\&#x200B;heit_**, but is returned as **_Ein&#x200B;heit_**
   This doesn't matter much in further processing, but it would be good to know that there is something in between **_Ein_** and **_heit_**

**Comment by nijel** [comment]
> How can I insert a zero width space in weblate's editor other than copy&pasting it from somewhere else?

You can add it to the [visual keyboard](https://docs.weblate.org/en/latest/user/translating.html#visual-keyboard).

**Reply by luebbe** [reply]
I added it to the visual keyboard via my user configuration. It works (on the keyboard), but since it is a ZWS, it's well hidden in the config.
![grafik](https://user-images.githubusercontent.com/3872576/128026155-42a4e774-36b0-451b-be3e-7d146bb5779c.png)

Is it possible to add special characters as unicode or xml escapes to the config? I tried adding `&#8203;` to the user config, but this adds `&`, `#`, ... to the visual keyboard. Is there a way to achieve this in a "visual" way?
The global config would be the next logical step, but I want to try it out first before bothering my admins with it.

**Reply by nijel** [reply]
You can also add these site-wide in settings.py (and use unicode escape sequences there).

**Comment by nijel** [comment, marked as answer]
> Would it be possible for weblate to highlight zero width spaces in the editor, like it does for linebreaks?

It should be possible, feel free to open issue for that.

**Reply by nijel** [reply]
Just for the record, this is covered by https://github.com/WeblateOrg/weblate/issues/6377

**Reply by luebbe** [reply]
I know, I wrote the ticket 😆

**Comment by nijel** [comment]
> Would it be possible for weblate to preserve this character as xml entity?

The serialization is implemented using [translate-toolkit](https://github.com/translate/translate/) and it's behavior varies depending on the file format.
